### PR TITLE
Feat remove redis

### PR DESCRIPTION
**Pull request by tofarr**
**End-user friendly description of the problem this fixes or functionality that this introduces**
This PR removes Redis dependency from OpenHands, simplifying the architecture and reducing operational complexity. The functionality previously handled by Redis has been reimplemented using a more streamlined approach with conversation managers.

- [x] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Key changes:
1. Removed Redis dependency completely
2. Introduced new conversation manager architecture:
   - Split SessionManager into standalone and clustered implementations
   - Created dedicated ConversationManager interface
   - Improved session handling and cleanup
3. Enhanced session management:
   - Better handling of session closing and cleanup
   - Improved file descriptor leak fixes
   - Added user ID propagation throughout the system
4. Improved stability:
   - More resilient error handling
   - Better cleanup of stale sessions
   - Fixed various edge cases in session management

The PR includes significant test coverage and fixes to ensure reliability.

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:ef3cd2e-nikolaik   --name openhands-app-ef3cd2e   docker.all-hands.dev/all-hands-ai/openhands:ef3cd2e
```

### CODE REVIEW

File: openhands/server/conversation_manager/standalone_conversation_manager.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+from openhands.utils.shutdown_listener import should_continue
+
+from .conversation_manager import ConversationManager
+
+_CLEANUP_INTERVAL = 15
+MAX_RUNNING_CONVERSATIONS = 3
+
+
+@dataclass
+class StandaloneConversationManager(ConversationManager):
```
Comment (rbren) [inline openhands/server/conversation_manager/standalone_conversation_manager.py:27]: It'd be nice to see the diff vs the old manager.py but maybe that's asking too much from git

Anything meaningful change here, other than removal of the redis stuff?

Comment (tofarr) [reply openhands/server/conversation_manager/standalone_conversation_manager.py:27]: The only changes should be:
* Rename of SessionManager -> ConversationManager.
* Split of SessionManager into 2 classes, removing all Redis Logic
* Using the `getImpl` process to get the ConversationManager (So we can use a different one in SAAS vs OSS)

Everything else should be functionally identical

**Review by rbren — APPROVED** [review]
